backend/src/main/java/org/eclipse/tractusx/puris/backend/demand/logic/adapter/ShortTermMaterialDemandSammMapper.java

```diff
@@ -0,0 +1,160 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.tractusx.puris.backend.demand.logic.adapter;
+
+import org.eclipse.tractusx.puris.backend.common.domain.model.measurement.ItemQuantityEntity;
+import org.eclipse.tractusx.puris.backend.demand.domain.model.DemandCategoryEnumeration;
+import org.eclipse.tractusx.puris.backend.demand.domain.model.OwnDemand;
+import org.eclipse.tractusx.puris.backend.demand.domain.model.ReportedDemand;
+import org.eclipse.tractusx.puris.backend.demand.logic.dto.demandsamm.Demand;
+import org.eclipse.tractusx.puris.backend.demand.logic.dto.demandsamm.DemandCategoryCharacteristic;
+import org.eclipse.tractusx.puris.backend.demand.logic.dto.demandsamm.DemandSeries;
+import org.eclipse.tractusx.puris.backend.demand.logic.dto.demandsamm.ShortTermMaterialDemand;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Material;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Partner;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialPartnerRelationService;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class ShortTermMaterialDemandSammMapper {
+    @Autowired
+    private MaterialPartnerRelationService mprService;
+
+    @Autowired
+    private MaterialService materialService;
+
+    public ShortTermMaterialDemand ownDemandToSamm(List<OwnDemand> demandList) {
+        if (demandList == null || demandList.isEmpty()) {
+            log.warn("Can't map empty list");
+            return null;
+        }
+        Partner partner = demandList.get(0).getPartner();
+        Material material = demandList.get(0).getMaterial();
+        if (demandList.stream().anyMatch(dem -> !dem.getPartner().equals(partner))) {
+            log.warn("Can't map demand list with different partners");
+            return null;
+        }
+
+        if (demandList.stream().anyMatch(prod -> !prod.getMaterial().equals(material))) {
+            log.warn("Can't map demand list with different materials");
+            return null;
+        }
+        var groupedByCategory = demandList
+                .stream()
+                .collect(Collectors.groupingBy(demand -> new DemandGroupingHelper(demand.getDemandCategoryCode(), demand.getDemandLocationBpns(), demand.getSupplierLocationBpns())));
```

Comment:
Wondering if we already check during creation of demands that there can just be one for the same day, category, location customer & supplier?

If not then, it's likely better to add this validation in the service.